pandas/tseries/resample.py

```diff
@@ -85,9 +85,14 @@ def __init__(self, obj, groupby=None, axis=0, kind=None, **kwargs):
         self.exclusions = set()
         self.binner = None
         self.grouper = None
+        self.from_selection = False
 
         if self.groupby is not None:
-            self.groupby._set_grouper(self._convert_obj(obj), sort=True)
+            # bookeeping to disallow upsampling if not resampling on index
+            self.from_selection = (self.groupby.key is not None or
```

Comment:
As noted in the discussion above, I'm disallowing upsampling if using a selection.  This is used to catch and raise an error.